req.txt

```diff
@@ -8,7 +8,7 @@ certifi==2020.12.5
 chardet==3.0.4
 click==7.1.2
 cycler==0.10.0
-dataclasses==0.8
+dataclasses==0.6
 Flask==1.1.2
 Flask-RESTful==0.3.8
 hazm==0.7.0
```

Comment:
please run command below to update your requirement file.
```
make freeze
```

Reply:
What does this command do? And I have pip3, and I change the makefile to pip3. Is that correct?

Reply:
No, don't change make file. It seems that you haven't used virtual environment to install pip packages. For every python project, we have to define a new virtual environement and install pip packages in that environment. For this purpose you have to run the following commands:
```
python3 -m pip install --user virtualenv
python3.6 -m venv .env
source .env/bin/activate # now you are in .env environemnt
pip install -r req.txt # to install packages in .env
#install extra packages. for example pip install websocket or every other package that you have added.
make freeze # to freeze all the packages that are present in .env to req.txt.
```
For more details, please visit https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/